Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -488,7 +489,14 @@ class TabTrayController: UIViewController, UITabBarDelegate, UICollectionViewDel
             } else {
                 cell.favicon.image = UIImage(named: "defaultFavicon")
             }
-            cell.background.image = tab.screenshot
+
+            if let img = tab.screenshot {
+                cell.background.image = img
+            } else if let thumbs = profile.thumbnails as? SDWebThumbnails {
+                let key = SDWebThumbnails.getKey(tab.url!)
+                cell.background.moz_getImageFromCache(key, cache: thumbs.cache, completed: { img, err, type, key in
+                })
+            }
```

Comment:
FWIW, @st3fan's recent change in [#491](https://github.com/mozilla/firefox-ios/pull/491/files#diff-221fd685cced62e8c7406c0b5f24ed6cR242) saves the screenshot in the session, so we shouldn't be losing screenshots anymore.